Sources/Charcoal/Filters/Map/MapFilterViewController.swift

```diff
@@ -2,84 +2,87 @@
+protocol MapFilterViewControllerDelegate: AnyObject {
+    func mapFilterViewController(_ mapFilterViewController: MapFilterViewController,
+                                 didSelect selection: CharcoalViewController.MapSelection)
 }
 
-final class MapFilterViewController: FilterViewController {
-    weak var mapDataSource: MapFilterDataSource? {
-        didSet {
-            if let mapTileOverlay = mapDataSource?.mapTileOverlay {
-                mapFilterView.setMapTileOverlay(mapTileOverlay)
-            }
-        }
-    }
+protocol ToggleFilter: AnyObject {
+    func resetFilterValues()
+    func updateFilterValues()
+}
+
+class MapFilterViewController: FilterViewController {
+    private let mapRadiusFilterViewController: MapRadiusFilterViewController
+    private let mapPolygonFilterViewController: MapPolygonFilterViewController?
+    private var selectedViewController: UIViewController
+
+    weak var mapFilterDelegate: MapFilterViewControllerDelegate?
 
     weak var searchLocationDataSource: SearchLocationDataSource? {
         didSet {
-            searchLocationViewController.searchLocationDataSource = searchLocationDataSource
+            mapRadiusFilterViewController.searchLocationDataSource = searchLocationDataSource
+            mapPolygonFilterViewController?.searchLocationDataSource = searchLocationDataSource
         }
     }
 
-    // MARK: - Private properties
-
-    private let latitudeFilter: Filter
-    private let longitudeFilter: Filter
-    private let radiusFilter: Filter
-    private let locationNameFilter: Filter
-    private let locationManager = CLLocationManager()
-    private var hasRequestedLocationAuthorization = false
-    private var nextRegionChangeIsFromUserInteraction = false
-    private var hasChanges = false
-    private var isMapLoaded = false
-
-    private lazy var mapFilterView: MapFilterView = {
-        let mapFilterView = MapFilterView(radius: radius, centerCoordinate: coordinate)
-        mapFilterView.translatesAutoresizingMaskIntoConstraints = false
-        mapFilterView.searchBar = searchLocationViewController.searchBar
-        mapFilterView.locationName = locationName
-        mapFilterView.delegate = self
-        return mapFilterView
+    private lazy var toggleButton: UIBarButtonItem = {
+        let buttonItem = UIBarButtonItem(
+            title: nil,
+            style: .plain,
+            target: self,
+            action: #selector(toggleViewControllers)
+        )
+        buttonItem.setTitleTextAttributes([.font: UIFont.bodyStrong], for: .normal)
+        return buttonItem
     }()
 
-    private lazy var searchLocationViewController: SearchLocationViewController = {
-        let searchLocationViewController = SearchLocationViewController()
-        searchLocationViewController.delegate = self
-        return searchLocationViewController
+    private lazy var mapContainerView: UIView = {
+        let view = UIView(frame: CGRect(x: 0, y: 0, width: 200, height: 200))
```

Comment:
When I don't init this with a height, I get a lot of breaking layout constraint issues when opening map search, because the map has size of zero when setting up the views. Is it okay to do this, or is there another solution?

Reply:
I think it is OK, I do it myself to avoid these issues. Very hard to work with 0 size views 😄